test/unit/bucketfs_test.py

```diff
@@ -0,0 +1,230 @@
+    _ = bucket[name]
+    assert bucket_mock.download.called
+    assert call(name, 8192) in bucket_mock.download.mock_calls
+
+
+def test_mapped_bucket_supports_item_access_based_upload():
+    bucket_mock = Mock(Bucket)
+    bucket = MappedBucket(bucket_mock)
+    name, data = "foo", bytes([1, 2, 3, 4])
+    bucket[name] = data
+
+    assert bucket_mock.upload.called
+    assert call(path=name, data=data) in bucket_mock.upload.mock_calls
+
+
+def test_mapped_bucket_supports_item_deletion_based_delete():
+    bucket_mock = Mock(Bucket)
+    bucket = MappedBucket(bucket_mock)
+    path = "/some/odd/path.txt"
+    del bucket[path]
+
+    assert bucket_mock.delete.called
+    assert call(path=path) in bucket_mock.delete.mock_calls
+
+
+@pytest.mark.parametrize(
+    "chunks,expected",
+    [
+        ((b"12" for _ in range(0, 4)), b"12121212"),
+        ([b"12" for _ in range(0, 4)], b"12121212"),
+        ((b for b in b"123457689"), b"123457689"),
+        ([b for b in b"123457689"], b"123457689"),
+    ],
+)
+def test_as_bytes_converter(chunks, expected):
+    assert as_bytes(chunks) == expected
+
+
+@pytest.mark.parametrize(
+    "chunks,expected",
+    [
+        ((b"12" for _ in range(0, 4)), "12121212"),
+        ([b"12" for _ in range(0, 4)], "12121212"),
+        ((b for b in b"123457689"), "123457689"),
+        ([b for b in b"123457689"], "123457689"),
+    ],
+)
+def test_as_string_converter(chunks, expected):
+    assert as_string(chunks) == expected
+
+
+@pytest.mark.parametrize(
+    "chunks,algorithm,expected",
+    [
+        (
+            (b"12" for _ in range(0, 4)),
+            "sha1",
+            "cc4723995ce819915e734147a77850427a9e95f9",
+        ),
+        ((b"12" for _ in range(0, 4)), "md5", "8ce87b8ec346ff4c80635f667d1592ae"),
+        (
+            [b"12" for _ in range(0, 4)],
+            "sha1",
+            "cc4723995ce819915e734147a77850427a9e95f9",
+        ),
+        ([b"12" for _ in range(0, 4)], "md5", "8ce87b8ec346ff4c80635f667d1592ae"),
+        ((b for b in b"123456789"), "sha1", "f7c3bc1d808e04732adf679965ccc34ca7ae3441"),
+        ((b for b in b"123456789"), "md5", "25f9e794323b453885f5181f1b624d0b"),
+        (
+            sorted([b for b in b"123457689"]),
+            "sha1",
+            "f7c3bc1d808e04732adf679965ccc34ca7ae3441",
+        ),
+        (sorted([b for b in b"123457689"]), "md5", "25f9e794323b453885f5181f1b624d0b"),
+    ],
+)
+def test_as_hash_converter(chunks, algorithm, expected):
+    assert as_hash(chunks, algorithm).hex() == expected
+
+
+@pytest.mark.parametrize(
+    "chunks,filename,expected_content",
+    [
+        ((b"12" for _ in range(0, 4)), "file.txt", b"12121212"),
+        ([b"12" for _ in range(0, 4)], "file.txt", b"12121212"),
+        ((b for b in b"123456789"), Path("file.txt"), b"123456789"),
+        ([b for b in b"123456789"], Path("file.txt"), b"123456789"),
+    ],
+)
+def test_as_file_converter(temporary_directory, chunks, filename, expected_content):
```

Comment:
is temporary_directory from pytest, I used so far tmp_path, is there a difference?
https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture